db-migrate doesn't compile using the new node 4.0.0

Crashes when installing sqlite3

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26573586-db-migrate-doesn-t-compile-using-the-new-node-4-0-0?utm_campaign=plugin&utm_content=tracker%2F73887&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F73887&utm_medium=issues&utm_source=github).
</bountysource-plugin>

Seeing this as well. Here's the output of `npm install db-migrate`:

```
> sqlite3@3.0.10 install /root/__test__/node_modules/db-migrate/node_modules/sqlite3
> node-pre-gyp install --fallback-to-build

make: Entering directory `/root/__test__/node_modules/db-migrate/node_modules/sqlite3/build'
  ACTION deps_sqlite3_gyp_action_before_build_target_unpack_sqlite_dep Release/obj/gen/sqlite-autoconf-3081101/sqlite3.c
  TOUCH Release/obj.target/deps/action_before_build.stamp
  CC(target) Release/obj.target/sqlite3/gen/sqlite-autoconf-3081101/sqlite3.o
Release/obj/gen/sqlite-autoconf-3081101/sqlite3.c: In function ‘rtreeInsertCell’:
Release/obj/gen/sqlite-autoconf-3081101/sqlite3.c:155150:26: warning: array subscript is above array bounds [-Warray-bounds]
       aCenterCoord[iDim] += DCOORD(aCell[ii].aCoord[iDim*2]);
                          ^
Release/obj/gen/sqlite-autoconf-3081101/sqlite3.c:155151:26: warning: array subscript is above array bounds [-Warray-bounds]
       aCenterCoord[iDim] += DCOORD(aCell[ii].aCoord[iDim*2+1]);
                          ^
Release/obj/gen/sqlite-autoconf-3081101/sqlite3.c:155150:26: warning: array subscript is above array bounds [-Warray-bounds]
       aCenterCoord[iDim] += DCOORD(aCell[ii].aCoord[iDim*2]);
                          ^
Release/obj/gen/sqlite-autoconf-3081101/sqlite3.c:155151:26: warning: array subscript is above array bounds [-Warray-bounds]
       aCenterCoord[iDim] += DCOORD(aCell[ii].aCoord[iDim*2+1]);
                          ^
Release/obj/gen/sqlite-autoconf-3081101/sqlite3.c:155150:26: warning: array subscript is above array bounds [-Warray-bounds]
       aCenterCoord[iDim] += DCOORD(aCell[ii].aCoord[iDim*2]);
                          ^
Release/obj/gen/sqlite-autoconf-3081101/sqlite3.c:155151:26: warning: array subscript is above array bounds [-Warray-bounds]
       aCenterCoord[iDim] += DCOORD(aCell[ii].aCoord[iDim*2+1]);
                          ^
Release/obj/gen/sqlite-autoconf-3081101/sqlite3.c:155150:26: warning: array subscript is above array bounds [-Warray-bounds]
       aCenterCoord[iDim] += DCOORD(aCell[ii].aCoord[iDim*2]);
                          ^
Release/obj/gen/sqlite-autoconf-3081101/sqlite3.c:155151:26: warning: array subscript is above array bounds [-Warray-bounds]
       aCenterCoord[iDim] += DCOORD(aCell[ii].aCoord[iDim*2+1]);
                          ^
  AR(target) Release/obj.target/deps/sqlite3.a
  COPY Release/sqlite3.a
  CXX(target) Release/obj.target/node_sqlite3/src/database.o
In file included from ../src/database.h:11:0,
                 from ../src/database.cc:5:
../node_modules/nan/nan.h:261:25: error: redefinition of ‘template<class T> v8::Local<T> _NanEnsureLocal(v8::Local<T>)’
 NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Local<T> val) {
                         ^
../node_modules/nan/nan.h:256:25: error: ‘template<class T> v8::Local<T> _NanEnsureLocal(v8::Handle<T>)’ previously declared here
 NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Handle<T> val) {
                         ^
../node_modules/nan/nan.h:661:13: error: ‘node::smalloc’ has not been declared
     , node::smalloc::FreeCallback callback
             ^
../node_modules/nan/nan.h:661:35: error: expected ‘,’ or ‘...’ before ‘callback’
     , node::smalloc::FreeCallback callback
                                   ^
../node_modules/nan/nan.h: In function ‘v8::Local<v8::Object> NanNewBufferHandle(char*, size_t, int)’:
../node_modules/nan/nan.h:665:50: error: ‘callback’ was not declared in this scope
         v8::Isolate::GetCurrent(), data, length, callback, hint);
                                                  ^
../node_modules/nan/nan.h:665:60: error: ‘hint’ was not declared in this scope
         v8::Isolate::GetCurrent(), data, length, callback, hint);
                                                            ^
../node_modules/nan/nan.h: In function ‘v8::Local<v8::Object> NanNewBufferHandle(const char*, uint32_t)’:
../node_modules/nan/nan.h:672:67: error: call of overloaded ‘New(v8::Isolate*, const char*&, uint32_t&)’ is ambiguous
     return node::Buffer::New(v8::Isolate::GetCurrent(), data, size);
                                                                   ^
../node_modules/nan/nan.h:672:67: note: candidates are:
In file included from ../node_modules/nan/nan.h:25:0,
                 from ../src/database.h:11,
                 from ../src/database.cc:5:
/root/.node-gyp/4.0.0/include/node/node_buffer.h:31:40: note: v8::MaybeLocal<v8::Object> node::Buffer::New(v8::Isolate*, v8::Local<v8::String>, node::encoding) <near match>
 NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                        ^
/root/.node-gyp/4.0.0/include/node/node_buffer.h:31:40: note:   no known conversion for argument 3 from ‘uint32_t {aka unsigned int}’ to ‘node::encoding’
/root/.node-gyp/4.0.0/include/node/node_buffer.h:43:40: note: v8::MaybeLocal<v8::Object> node::Buffer::New(v8::Isolate*, char*, size_t) <near match>
 NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                        ^
/root/.node-gyp/4.0.0/include/node/node_buffer.h:43:40: note:   no known conversion for argument 2 from ‘const char*’ to ‘char*’
In file included from ../src/database.h:11:0,
                 from ../src/database.cc:5:
../node_modules/nan/nan.h: In function ‘v8::Local<v8::Object> NanNewBufferHandle(uint32_t)’:
../node_modules/nan/nan.h:676:61: error: could not convert ‘node::Buffer::New(v8::Isolate::GetCurrent(), ((size_t)size))’ from ‘v8::MaybeLocal<v8::Object>’ to ‘v8::Local<v8::Object>’
     return node::Buffer::New(v8::Isolate::GetCurrent(), size);
                                                             ^
../node_modules/nan/nan.h: In function ‘v8::Local<v8::Object> NanBufferUse(char*, uint32_t)’:
../node_modules/nan/nan.h:683:12: error: ‘Use’ is not a member of ‘node::Buffer’
     return node::Buffer::Use(v8::Isolate::GetCurrent(), data, size);
            ^
../src/database.cc: In static member function ‘static void node_sqlite3::Database::Work_BeginOpen(node_sqlite3::Database::Baton*)’:
../src/database.cc:145:9: warning: unused variable ‘status’ [-Wunused-variable]
     int status = uv_queue_work(uv_default_loop(),
         ^
../src/database.cc: In static member function ‘static void node_sqlite3::Database::Work_BeginClose(node_sqlite3::Database::Baton*)’:
../src/database.cc:230:9: warning: unused variable ‘status’ [-Wunused-variable]
     int status = uv_queue_work(uv_default_loop(),
         ^
../src/database.cc: In static member function ‘static void node_sqlite3::Database::Work_BeginExec(node_sqlite3::Database::Baton*)’:
../src/database.cc:509:9: warning: unused variable ‘status’ [-Wunused-variable]
     int status = uv_queue_work(uv_default_loop(),
         ^
../src/database.cc: In static member function ‘static void node_sqlite3::Database::Work_BeginLoadExtension(node_sqlite3::Database::Baton*)’:
../src/database.cc:610:9: warning: unused variable ‘status’ [-Wunused-variable]
     int status = uv_queue_work(uv_default_loop(),
         ^
make: *** [Release/obj.target/node_sqlite3/src/database.o] Error 1
make: Leaving directory `/root/__test__/node_modules/db-migrate/node_modules/sqlite3/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 3.10.0-229.11.1.el7.x86_64
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/root/__test__/node_modules/db-migrate/node_modules/sqlite3/lib/binding/node-v46-linux-x64/node_sqlite3.node" "--module_name=node_sqlite3" "--module_path=/root/__test__/node_modules/db-migrate/node_modules/sqlite3/lib/binding/node-v46-linux-x64"
gyp ERR! cwd /root/__test__/node_modules/db-migrate/node_modules/sqlite3
gyp ERR! node -v v4.0.0
gyp ERR! node-gyp -v v3.0.1
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/root/__test__/node_modules/db-migrate/node_modules/sqlite3/lib/binding/node-v46-linux-x64/node_sqlite3.node --module_name=node_sqlite3 --module_path=/root/__test__/node_modules/db-migrate/node_modules/sqlite3/lib/binding/node-v46-linux-x64' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/root/__test__/node_modules/db-migrate/node_modules/sqlite3/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:87:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:817:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
node-pre-gyp ERR! System Linux 3.10.0-229.11.1.el7.x86_64
node-pre-gyp ERR! command "/usr/local/bin/node" "/root/__test__/node_modules/db-migrate/node_modules/sqlite3/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /root/__test__/node_modules/db-migrate/node_modules/sqlite3
node-pre-gyp ERR! node -v v4.0.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.9
node-pre-gyp ERR! not ok 
Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/root/__test__/node_modules/db-migrate/node_modules/sqlite3/lib/binding/node-v46-linux-x64/node_sqlite3.node --module_name=node_sqlite3 --module_path=/root/__test__/node_modules/db-migrate/node_modules/sqlite3/lib/binding/node-v46-linux-x64' (1)
npm ERR! Linux 3.10.0-229.11.1.el7.x86_64
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "db-migrate"
npm ERR! node v4.0.0
npm ERR! npm  v2.14.2
npm ERR! code ELIFECYCLE

npm ERR! sqlite3@3.0.10 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sqlite3@3.0.10 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! This is most likely a problem with the sqlite3 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get their info via:
npm ERR!     npm owner ls sqlite3
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /root/__test__/npm-debug.log
```

--- a/package.json
+++ b/package.json
@@ -68,7 +68,7 @@
     "proxyquire": "^1.4.0",
     "rimraf": "~2.4.0",
     "sinon": "^1.14.1",
-    "sqlite3": "^3.0.11",
+    "sqlite3": "^3.1.0",
     "vows": "0.8.0"
   },
   "scripts": {

--- a/package.json
+++ b/package.json
@@ -68,7 +68,7 @@
     "proxyquire": "^1.4.0",
     "rimraf": "~2.4.0",
     "sinon": "^1.14.1",
-    "sqlite3": "^3.0.10",
+    "sqlite3": "^3.0.11",
     "vows": "0.8.0"
   },
   "scripts": {

